src/recipes/recipes.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  Query,
+} from '@nestjs/common';
+import { RecipesService } from './recipes.service';
+import { CreateRecipeDto } from './dto/create-recipe.dto';
+import { UpdateRecipeDto } from './dto/update-recipe.dto';
+import { GetUserId } from 'src/common/decorators/getUserId.decorator';
+import { GetRecipesQueryParamsDto } from './dto/get-recipe-query-params.dto';
+import { GetRecipesResponse } from './dto/get-recipe-response';
+
+@Controller('recipes')
+export class RecipesController {
+  constructor(private readonly recipesService: RecipesService) {}
+
+  @Post()
+  create(@Body() createRecipeDto: CreateRecipeDto) {
+    return this.recipesService.create(createRecipeDto);
+  }
+
+  @Get()
+  getRecipes(
+    @GetUserId() userId: string | null,
+    @Query() queryParams: GetRecipesQueryParamsDto,
+  ): Promise<GetRecipesResponse> {
+    return this.recipesService.getRecipes(userId, queryParams);
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.recipesService.findOne(+id);
+  }
+
+  @Patch(':id')
+  update(@Param('id') id: string, @Body() updateRecipeDto: UpdateRecipeDto) {
+    return this.recipesService.update(+id, updateRecipeDto);
+  }
+
+  @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.recipesService.remove(+id);
+  }
```

Comment:
you are using swagger annotations in DTO's but you haven't used any swagger annotation on the endpoints